oxidation/crates/parsec_api_crypto/src/sodiumoxide/secret.rs

```diff
@@ -31,15 +31,17 @@ impl SecretKey {
     }
 
     pub fn decrypt(&self, ciphered: &[u8]) -> Result<Vec<u8>, &'static str> {
-        let nonce_slice = ciphered.get(..NONCEBYTES).ok_or("Invalid data size")?;
+        let nonce_slice = ciphered
+            .get(..NONCEBYTES)
+            .ok_or("The nonce must be exactly 24 bytes long")?;
```

Comment:
Not sure we want to explain too much on why the decryption failed: data are either right or totally broken so the caller should not be tempted to return a detailed status that could help an attacker...

What do you think ?

Reply:
we can't change this error message as long as cryptoBox send back this error